cluster/src/main/java/io/servicefabric/cluster/NetworkTimer.java

```diff
@@ -13,20 +13,35 @@
 
 import com.google.common.base.Preconditions;
 
-class Timer {
+/**
+ * It is an utility wrapper class around {@code io.netty.util.HashedWheelTimer} which simplify its usage.
+ * This timer doesn't execute scheduled job on time, but provide approximate execution and on each tick
+ * checks either there are any jobs behind the timer and execute them. In most network applications,
+ * I/O timeout does not need to be accurate. The default tick duration is 100 milliseconds.
+ *
+ * @see io.netty.util.HashedWheelTimer
+ */
+class NetworkTimer {
```

Comment:
Because of this comment
  This timer doesn't execute scheduled job on time ... In most network applications,
  I/O timeout does not need to be accurate. The default tick duration is 100 milliseconds.

But I agree - not a best name - do you have better ideas?